feat(buttonGroup): new component

wbarbee opened the pull request:
## Summary

New `ButtonGroup` component nested within the `button` directory and building off of the existing Button component.

## ADO Story or GitHub Issue Link

[ADO User Story](https://dev.azure.com/Kyndryl/Shidoka%20-%20Bridge%20Design%20System/_workitems/edit/2360117)

## Figma Link

[Figma Link](https://www.figma.com/design/qyPEUQckxj8LUgesi1OEES/Component-Library-2.0?node-id=35960-7018&m=dev)

## Checklist

- [ ] Used Conventional Commit messages as outlined in the contributing guide.
  - [ ] Noted breaking changes (if any).
- [ ] Documented/updated all props, events, slots, parts, etc with JSDoc.
  - [ ] Ran the `analyze` command to update Storybook docs.
- [ ] Added/updated Stories with controls to cover all variants.
- [ ] Ran `test` locally to address any failures.
- [ ] Added/updated the Figma link for the Story's Design tab.
- [ ] Added any new component exports to the src/index.ts file

## Screenshots

*when available*

kumarivijeta commented:
Few Observation:

1. I thing `kyn-button-group` should not be nested inside Button component.
2.
<img width="816" height="319" alt="image" src="https://github.com/user-attachments/assets/846b7595-f2d3-40b2-8a1c-465bf01c18ae" />

3. On click of any button in button group event is getting emitted 3 times.
    https://deploy-preview-532--shidoka-applications.netlify.app/?path=/story/components-button-button-group--default
    https://deploy-preview-532--shidoka-applications.netlify.app/?path=/story/components-button-button-group--icons
  
4. On Tabbing focus is not very much visible 

<img width="478" height="124" alt="image" src="https://github.com/user-attachments/assets/3e22dbd5-5a93-46eb-a30f-dffdb815fe2c" />

kumarivijeta commented on src/components/reusable/button/buttonGroup.stories.js:
```diff
@@ -0,0 +1,236 @@
+};
+
+const IconTemplate = (args) => {
+  const [, updateArgs] = useArgs();
```
Just curious, What is extra comma(,) does here ?

wbarbee commented:
1. moved it to it's own component
2. fixed
3. fixed
4. I went through a couple of options with Marco yesterday afternoon and this is what we arrived at, removing the outline-offset and increasing the z-index, as it says in the notes above. @brian-patrick-3, we can change the color of the focus outline for these grouped buttons to make the contrast a bit better?

brian-patrick-3 commented:
> 4. I went through a couple of options with Marco yesterday afternoon and this is what we arrived at, removing the outline-offset and increasing the z-index, as it says in the notes above. @brian-patrick-3, we can change the color of the focus outline for these grouped buttons to make the contrast a bit better?

Yeah it definitely makes sense to use a different focus color in this context.

wbarbee commented on src/components/reusable/buttonGroup/buttonGroup.ts:
```diff
@@ -0,0 +1,353 @@
+
+      const handler = () => this._onButtonClick(idx);
+      this._boundHandlers.set(btn, handler);
+      btn.addEventListener('click', handler);
```
doesn't `@click` listen for native browser clicks, while `@on-click` listen for a custom event called "on-click"?

brian-patrick-3 replied:
Right, generally we want to interface with the custom events emitted from our components.

wbarbee commented on src/components/reusable/buttonGroup/buttonGroup.ts:
```diff
@@ -0,0 +1,353 @@
+
+  /** Button group kind */
+  @property({ type: String })
+  accessor kind: BUTTON_GROUP_KINDS = BUTTON_GROUP_KINDS.DEFAULT;
```
✅

brian-patrick-3 commented on src/components/reusable/buttonGroup/buttonGroup.ts:
```diff
@@ -0,0 +1,353 @@
+  private _emitChange(value: unknown) {
+    this.dispatchEvent(
+      new CustomEvent('on-change', {
+        bubbles: true,
+        composed: true,
```
Right, without bubbling they have to place the event listener on the component tag itself. Generally I only bubble if we need to capture custom events internally from slotted children. I did have one instance where bubbling was used in nested elements which caused a problem, so I try to be careful about it.

wbarbee replied:
removed both `bubbles: true`

brian-patrick-3 commented on src/components/reusable/buttonGroup/buttonGroup.stories.js:
```diff
@@ -0,0 +1,231 @@
+    const [, updateArgs] = useArgs();
+    return html`
+      <div class="helper-label">
+        Page ${args.selectedIndex + 1} of ${args.totalPages} — showing
```
Oh sorry, yeah it's fine in that case

brian-patrick-3 commented:
:tada: This PR is included in version 2.51.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@kyndryl-design-system/shidoka-applications/v/2.51.0)
- [GitHub release](https://github.com/kyndryl-design-system/shidoka-applications/releases/tag/v2.51.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
